### ISSUE TITLE
i18n-t component uses incorrect scope when used in slot

### ISSUE THREAD

**Issue by stephenhebert** [intlify/vue-i18n-next#1486]
### Reporting a bug?

Normally, when using `<i18n-t>` component, the scope will default to local SFC scope.  However, when using `<i18n-t>` component within HeadlessUI's component, the global scope will be used.  (This differs from the scope of the `t` function, which will continue to use local SFC scope as expected.)

### Expected behavior

It is expected that the scope of `<i18n-t>` component would be the local SFC component scope, as consistent with the `t` function.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-dl6wva?file=vite.config.ts

### System Info

```shell
System:
    OS: macOS 13.4.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 77.98 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.0 - ~/.nvm/versions/node/v16.20.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.0/bin/npm
    pnpm: 8.6.5 - ~/.nvm/versions/node/v16.20.0/bin/pnpm
  Browsers:
    Chrome: 115.0.5790.170
    Safari: 16.5.1
```

### Screenshot

<img width="504" alt="image" src="https://github.com/intlify/vue-i18n-next/assets/6313699/815f6eed-17c8-476f-aecf-8557f004bc88">

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by stephenhebert** [comment]
@RobinMalfait from Headless UI did some troubleshooting and figured out that this issue isn't related to Headless UI at all, but is a matter of Vue-i18n's scope resolution when using slots.

Here's an updated reproduction: https://stackblitz.com/edit/vitejs-vite-dl6wva?file=src%2Fcomponents%2FBaseSlot.vue

In short, if the `i18n-t` component is used within a slot for a component that also implements vue-i18n and that component also has a matching key, the `i18n-t` component will use that slotted component's scope to resolve the translation.  If the component does not have a matching key, it will use the global scope to resolve the translation.  If vue-i18n is not implemented in the slotted component, it will use the parent component's scope as expected.

Note: this is only an issue with the `i18n-t` component.

**Comment by stephenhebert** [comment]
I think this may be a duplicate of https://github.com/intlify/vue-i18n-next/issues/1392.  I've confirmed that the issue is resolved in the latest 9.3.0-beta. 🙏
